kernel/mempool.c

```diff
@@ -113,6 +113,40 @@ void *k_calloc(size_t nmemb, size_t size)
 	return ret;
 }
 
+void *k_realloc(void *ptr, size_t size)
+{
+	struct k_heap *heap, **heap_ref;
+	void *ret;
+
+	if (size == 0) {
+		k_free(ptr);
+		return NULL;
+	}
+	if (ptr == NULL) {
+		return k_malloc(size);
+	}
+	heap_ref = ptr;
+	ptr = --heap_ref;
+	heap = *heap_ref;
+
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap_sys, k_realloc, heap, ptr);
+
+	if (size_add_overflow(size, sizeof(heap_ref), &size)) {
+		SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_heap_sys, k_realloc, heap, ptr, NULL);
+		return NULL;
+	}
+
+	ret = k_heap_realloc(heap, ptr, size, K_NO_WAIT);
+
+	SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_heap_sys, k_realloc, heap, ptr, ret);
+
+	if (ret != NULL) {
+		heap_ref = ret;
```

Comment:
Sorry, I am not familiar with memory system.
Can it be `ret ++` ?

Reply:
I tested it:
```
/home/fin/zephyrproject/zephyr/kernel/mempool.c: In function 'k_realloc':
/home/fin/zephyrproject/zephyr/kernel/mempool.c:142:20: error: wrong type argument to increment [-Werror=pointer-arith]
  142 |                 ret++;
      |                    ^~
```
Unfortunately not

Reply:
We can't perform arithmetic on a void pointer.

Reply:
> We can't perform arithmetic on a void pointer.

Oh, thanks for your information.